text/0000-cargo-plugin-fields.md

```diff
@@ -0,0 +1,119 @@
+```
+"@EO13526.clearance = { classification = "unclasified", history = [...]}
+```
+
+So because cargo ignores the top @EO13526 it also ignores classification and history sub-fields.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There already exists the package.metadata table, it doesn't allow in place fields, but does allow adding of fields unrecognized by cargo.
+
+And does so without an ugly prefix and key naming convention.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- When a cargo plugin is providing it's own build command, and said build requires additional metadata in order to function.
+
+    It is desirable to have said metadata in place.
+- The choice of `@` for the plugin prefix was chosen over `_`, and `.`,
+
+    Dot: being the path separator for toml does not compose well in keypaths, with keys like `package..EO13526`, one of the dots being a separator for a keypath and one the name of the key.
+
+    Underscore: Could work well e.g. `package._EO13526.classification`,
+    where "_EO13526.classification" is a key.
+    This has the benefit of being similar to rusts identifier naming behavior. 
+
+    The other benefit of this choice is that `_` is not reqired to be quoted by toml.
+
+    At-Sign: Was chosen for the proposal because it stresses that the key is being direct _at_ the plugin, rather than an implementation detail that cargo itself ignores the field.
+- Choice of in field separator this proposal currently outlines using Dot as the in field separator between plugin name and key.
+    Alternative choices:
+
+    Colon: such as "@pluginname:key"  This has the benefit of being compatible with the json-ld specification.
+    Since toml documents are valid JSON documents, we could perhaps convert a Cargo.toml document by translating
+    the existing fields into "@Cargo:package", and convert Cargo.toml into json-ld and back.
+
+    This seems like an interesting idea worth exploring, in that it brings along the potential checking the validity of a Cargo.toml containing plugin fields.
+
+# Prior art
+[prior-art]: #prior-art
+
+From build system perspective, Cargo is fairly unique in it's behavior of not allowing arbitrary keys in-place in build configuration files.
+
+- [json-ld](https://www.w3.org/TR/json-ld/): Is somewhat similiar if not more structured, it supports contexts of the style `@prefix:suffix`.
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Whether we should reserve a prefix at all
+- Iff so what is the best prefix to reserve
+- Currently does not focus on how Cargo should present the fields to plugins, whether Cargo should filter out keys that do not resolve to the plugins name.  Or if plugins are allowed to implement behaviors based upon the keys of other plugins.
+
+
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+It would be really nice if in the future there was a `map` behavior
```

Comment:
I can’t figure out from this description or your code what this is supposed to mean.